### PR TITLE
core(user-timings): move into computed artifact

### PR DESCRIPTION
**Pull request by midzer**
Hi again :)

**Summary**
What kind of change does this PR introduce?
Move logic to fill userTimings Array into respective file in computed folder

Is this a bugfix, feature, refactoring, build related change, etc?
Internal refactoring

Describe the need for this change
Allow wider access to data for custom audits

**Present Issues**
Right now, there are still issues in `artifacts.d.ts` to find `MarkEvent` and `MeasureEvent` types. I don't know how to solve this as putting a `/** @typedef` in file's head doesn't seem to help (looked it up in `/audits/user-timings.js`)

**Related Issues/PRs**

https://github.com/GoogleChrome/lighthouse/issues/6696

Have a nice week
midzer

### CODE REVIEW

**Comment by midzer** [comment]
Thanks for your feedback, @paulirish 

After fixing type checks, there's still something going wrong with our tests on LH extension:
```
1) Lighthouse chrome extension should contain all categories:
      AssertionError [ERR_ASSERTION]: all categories not found
      + expected - actual
      -[]
      +[
      +  "accessibility"
      +  "best-practices"
      +  "performance"
      +  "pwa"
      +  "seo"
      +]
```
Maybe some1 has more insight on this error.

File: types/artifacts.d.ts [hunk 1]
```diff
@@ -393,6 +393,15 @@ declare global {
          */
         fmpFellBack: boolean;
       }
+
+      export type UserTimings = {
```
Comment (patrickhulce) [inline types/artifacts.d.ts]: isn't this supposed to be an array of user timings not a keyed object?

Comment (midzer) [reply types/artifacts.d.ts]: This way the only way I managed to get type checking done after a while.
If you have an easier solution, please let me know :)

File: lighthouse-core/computed/user-timings.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+/** @typedef {{name: string, isMark: false, args: LH.TraceEvent['args'], startTime: number, endTime: number, duration: number}} MeasureEvent */
+
+class UserTimings {
+  /**
+   * @param {LH.Trace} trace
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Artifacts.UserTimings>}
+   */
+  static async compute_(trace, context) {
+    return TraceOfTab.request(trace, context).then(traceOfTab => {
```
[the remark this answers is not in the document]
Comment (midzer) [reply lighthouse-core/computed/user-timings.js]: Great idea. Just pushed to branch :]

**Pull request merged**